planner/core/task.go

```diff
@@ -1056,12 +1061,21 @@ func (p *PhysicalTopN) pushTopNDownToDynamicPartition(copTsk *copTask) (task, bo
 		}
 		finalTblScanPlan := copTsk.tablePlan
 		for len(finalTblScanPlan.Children()) > 0 {
+			selOnTblScan, _ = finalTblScanPlan.(*PhysicalSelection)
 			finalTblScanPlan = finalTblScanPlan.Children()[0]
 		}
 		tblScan = finalTblScanPlan.(*PhysicalTableScan)
 		tblInfo = tblScan.Table
 	}
 
+	// Note that we only need to care about one Selection at most.
```

Comment:
only care about specified tree type?

Reply:
Yes. From the current implementation, the only thing that could possibly be here is a `Selection`.
@winoros can you verify this?

Reply:
yes